packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -327,6 +341,83 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+/**
+ * Sets up the picture in picture controller and assigns the AVPictureInPictureControllerDelegate to
+ * the controller.
+ */
+- (void)setupPiPController {
+  if (@available(macOS 10.15, *)) {
+    if ([AVPictureInPictureController isPictureInPictureSupported]) {
+      self.pictureInPictureController =
+          [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+      [self setAutomaticallyStartsPictureInPicture:NO];
+      self.pictureInPictureController.delegate = self;
+    }
+  } else {
+    // We don't do anything here because there is no setup required below macOS 10.15.
+  }
+}
+
+- (void)setAutomaticallyStartsPictureInPicture:
+    (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (!self.pictureInPictureController) return;
+#if TARGET_OS_IOS
+  if (@available(iOS 14.2, *)) {
+    self.pictureInPictureController.canStartPictureInPictureAutomaticallyFromInline =
+        canStartPictureInPictureAutomaticallyFromInline;
+  }
+#endif
+}
+
+- (void)setPictureInPictureOverlaySettings:(CGRect)frame {
+  if (_player) {
```

Comment:
Please don't mix `_foo` and `self.foo` within the same method. Also, this check isn't necessary.

Reply:
At which point this whole method is a one-liner whose implementation is shorter than the signature of the wrapper method, so you should just inline the code below and remove the wrapper.